src/main/java/org/openrewrite/java/migrate/trait/ScopedVariable.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.trait;
+
+import lombok.RequiredArgsConstructor;
+import lombok.Value;
+import org.jspecify.annotations.Nullable;
+import org.openrewrite.Cursor;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.trait.SimpleTraitMatcher;
+import org.openrewrite.trait.Trait;
+
+@Value
+public class ScopedVariable implements Trait<J.VariableDeclarations.NamedVariable> {
+    Cursor cursor;
+    Cursor scope;
+    J.Identifier identifier;
+
+    @RequiredArgsConstructor
+    public static class Matcher extends SimpleTraitMatcher<ScopedVariable> {
+
+        @Override
+        protected @Nullable ScopedVariable test(Cursor cursor) {
+            if (cursor.getValue() instanceof J.VariableDeclarations.NamedVariable) {
```

Comment:
We can likely just use variable.getDeclaringScope(cursor) to obtain the variable's scope; using a trait might be overkill.

A scenario where a trait could be useful, suppose I'm at an identifier and need to determine the declaring scope of the variable that this identifier references.

```
void foo() {
   int x = 5;
   if (true) {
      print(x) // find the declaring scope of x at this point,
   }
}
```